google-cloud-bigquery/lib/google/cloud/bigquery/dataset.rb

```diff
@@ -1709,16 +1711,21 @@ def load table_id, files, format: nil, create: nil, write: nil,
         #
         def reload!
           ensure_service!
-          reloaded_gapi = service.get_dataset dataset_id
+          @gapi = service.get_dataset dataset_id
           @reference = nil
-          @gapi = reloaded_gapi
+          @exists = nil
```

Comment:
Why is `@exists` set to `nil` here? If it successfully reloads without throwing an error shouldn't it be `true`?

Reply:
`#exists?` (without `force`) can return true *either* based on memoization (`@exists`) of a true result from a previous call, or based on the presence of the full resource representation (`@gapi`). Our decision was to clear the memoization state after a successful reload and rely on the full resource representation. It seems cleaner and easier to understand to me that way. I am not a fan of the memoization.

Reply:
Sounds good, thanks for the explanation!